File: LyricsShow_Test/LyricsShow_Test/ViewController.m
//
//  ViewController.m
//  LyricsShow_Test
//
//  Created by nali on 16/9/14.
//  Copyright © 2016年 Kimi. All rights reserved.
//

#import "ViewController.h"
#import "KMLyricsView.h"

@interface ViewController ()

@end

@implementation ViewController
{
    KMLyricsView *_lyricsView;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.view.backgroundColor = [UIColor blackColor];
    _lyricsView = [[KMLyricsView alloc] initWithFrame:CGRectMake(0.0, 0.0, [UIScreen mainScreen].bounds.size.width, 100.0)];
    _lyricsView.center = self.view.center;
    [self.view addSubview:_lyricsView];    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)clickChangeColors:(UIButton *)sender {
    CGFloat red = arc4random() / (CGFloat)INT_MAX;
    CGFloat green = arc4random() / (CGFloat)INT_MAX;
    CGFloat blue = arc4random() / (CGFloat)INT_MAX;
    
    UIColor *didReadColor = [UIColor colorWithRed:red green:green blue:blue alpha:1.0];
    UIColor *unreadColor = [UIColor colorWithRed:(1.0-red) green:(1.0-green) blue:(1.0-blue) alpha:1.0];
    _lyricsView.lyricsDidReadColor = didReadColor;
    _lyricsView.lyricsUnreadColor = unreadColor;
}

@end

File: TheLyrics/TheLyrics/ViewController.swift
//
//  ViewController.swift
//  TheLyrics
//
//  Created by saitjr on 8/8/16.
//  Copyright © 2016 saitjr. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        setupUI()
    }
}

extension ViewController {
    private func setupUI() {
        view.backgroundColor = .blackColor()
        
        let orangeView = UIView()
        orangeView.frame = CGRect(x: 0, y: 0, width: 100, height: 60)
        orangeView.center = CGPoint(x: orangeView.center.x, y: view.center.y)
        orangeView.backgroundColor = .orangeColor()
        view.addSubview(orangeView)
        
        let imageView = UIImageView(image: UIImage(named: "Text.png"))
        imageView.bounds = CGRect(x: 0, y: 0, width: view.bounds.size.width, height: 80)
        imageView.center = view.center
        imageView.contentMode = .ScaleAspectFit
        view.addSubview(imageView)
        
        UIView.animateWithDuration(3, delay: 0, options: [.Repeat, .Autoreverse],
            animations: {
                orangeView.frame.origin.x = self.view.bounds.size.width - orangeView.bounds.size.width
            }, completion: nil)
    }
}